lib/streamlit/runtime/app_session.py

```diff
@@ -356,6 +355,12 @@ def request_rerun(self, client_state: ClientState | None) -> None:
 
         if client_state:
             fragment_id = client_state.fragment_id
+            if fragment_id and not self._fragment_storage.contains(fragment_id):
+                _LOGGER.error(
+                    f"The fragment with id {fragment_id} does not exist anymore - "
+                    "it might have been removed during a preceding full-app rerun."
+                )
```

Comment:
I might worry about the implication of an error. Maybe a warning at worse, but perhaps even info? Especially if users can safely ignore the message, then we don't need to push this out as much.

Reply:
Good point, will change it!